app/models/location.rb

```diff
@@ -130,6 +130,14 @@ def validate_confirmed
     self.confirmed = [postcode, location_admin_district, location_code, units, type_of_unit, mobility_type].all?(&:present?)
   end
 
+  def linked_local_authorities
+    la = LocalAuthority.find_by(code: location_code)
+    return [] unless la
+
+    location_local_authorities = [la] + la.linked_local_authorities
+    LocalAuthority.where(id: location_local_authorities.map(&:id))
+  end
```

Comment:
Can this not be an association?

Reply:
hm, I'm not too sure, I tried to make it one, but we don't save the LA code as key in the links table and from location we only have a code. Maybe possible but not sure if clearer

Reply:
```suggestion
  def linked_local_authorities
    local_authority = LocalAuthority.where(code: location_code).pluck(:id)
    return local_authority if local_authority.empty?

    LocalAuthority.where(id: local_authority + la.linked_local_authorities_ids)
  end
```

Reply:
```suggestion
  def linked_local_authorities
    la = LocalAuthority.find_by(code: location_code)
    return [] unless la

    LocalAuthority.where(id: [la.id] + la.linked_local_authorities_ids)
  end
```

If you don't map the `linked_local_authorities` you can just pluck out the IDs. You can even do:

```ruby
LocalAuthority.where(id: la.linked_local_authorities_ids + la.id)
```

... if you don't mind about the ordering.